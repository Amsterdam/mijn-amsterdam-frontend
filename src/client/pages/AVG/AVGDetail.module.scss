.AvgStatusLines {
  margin-top: 4rem;
}
