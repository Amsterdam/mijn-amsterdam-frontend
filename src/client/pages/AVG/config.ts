import { AVGRequestFrontend } from '../../../server/services/avg/types';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

export const displayPropsAanvragen: DisplayProps<
  WithDetailLinkComponent<AVGRequestFrontend>
> = {
  detailLinkComponent: 'Nummer',
  ontvangstDatumFormatted: 'Ontvangen op',
  themas: 'Onderwerp(en)',
};

export const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => !avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
};

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/privacy/loket/',
    title: 'Loket persoonsgegevens gemeente Amsterdam',
  },
];
