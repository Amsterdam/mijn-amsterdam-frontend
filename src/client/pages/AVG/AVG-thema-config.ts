import { generatePath } from 'react-router-dom';

import { AVGRequestFrontend } from '../../../server/services/avg/types';
import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';
import { withOmitDisplayPropsForSmallScreens } from '../../components/Table/helpers';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';
import { MAX_TABLE_ROWS_ON_THEMA_PAGINA } from '../../config/app';
import { TrackingConfig } from '../../config/routes';

const MAX_TABLE_ROWS_ON_THEMA_PAGINA_LOPEND = 5;

const displayPropsAanvragenBase: DisplayProps<
  WithDetailLinkComponent<AVGRequestFrontend>
> = {
  detailLinkComponent: 'Nummer',
  ontvangstDatumFormatted: 'Ontvangen op',
  themas: 'Onderwerp(en)',
};

const displayPropsAanvragen = withOmitDisplayPropsForSmallScreens(
  displayPropsAanvragenBase,
  ['themas']
);

const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
} as const;

export type ListPageParamKey = keyof typeof listPageParamKind;
export type ListPageParamKind = (typeof listPageParamKind)[ListPageParamKey];

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
} as const;

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => !avgVerzoek.datumAfhandeling,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA_LOPEND,
    listPageRoute: generatePath(AppRoutes['AVG/LIST'], {
      kind: listPageParamKind.inProgress,
    }),
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => avgVerzoek.datumAfhandeling,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA,
    listPageRoute: generatePath(AppRoutes['AVG/LIST'], {
      kind: listPageParamKind.completed,
    }),
    ...tableConfigBase,
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/privacy/loket/',
    title: 'Loket persoonsgegevens gemeente Amsterdam',
  },
] as const;

export function getAVGListPageDocumentTitle(themaTitle: string) {
  return <T extends Record<string, string>>(
    config: TrackingConfig,
    params: T | null
  ) => {
    const kind = params?.kind as ListPageParamKind;
    return kind in tableConfig
      ? `${tableConfig[kind].title} | ${themaTitle}`
      : themaTitle;
  };
}
