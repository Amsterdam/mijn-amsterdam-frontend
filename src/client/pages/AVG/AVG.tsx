import { AppRoutes, ChapterTitles } from '../../../universal/config';
import { defaultDateFormat, isLoading } from '../../../universal/helpers';
import {
  addTitleLinkComponent,
  ChapterIcon,
  Linkd,
  OverviewPage,
  PageContent,
  PageHeading,
  SectionCollapsible,
  Table,
} from '../../components';
import { useAppStateGetter } from '../../hooks';
import styles from './AVG.module.scss';

const DISPLAY_PROPS_AVG = {
  idAsLink: 'Nummer',
  ontvangstDatum: 'Ontvangen op',
  themaString: 'Onderwerp',
};

const AVG = () => {
  const { AVG } = useAppStateGetter();

  const avgVerzoeken = AVG.content?.verzoeken?.map((avgVerzoek) => ({
    ...avgVerzoek,

    ontvangstDatum: defaultDateFormat(avgVerzoek.ontvangstDatum),
    idAsLink: avgVerzoek.id,
    themaString: avgVerzoek.themas.join(', '),
  }));

  const avgVerzoekenLopend = avgVerzoeken
    ? addTitleLinkComponent(
        avgVerzoeken.filter((avgVerzoek) => avgVerzoek.datumAfhandeling === ''),
        'idAsLink'
      )
    : [];
  const avgVerzoekenAfgehandeld = avgVerzoeken
    ? addTitleLinkComponent(
        avgVerzoeken.filter((avgVerzoek) => avgVerzoek.datumAfhandeling !== ''),
        'idAsLink'
      )
    : [];

  return (
    <OverviewPage className={styles.AVG}>
      <PageHeading
        backLink={{
          to: AppRoutes.HOME,
          title: 'Home',
        }}
        isLoading={isLoading(AVG)}
        icon={<ChapterIcon />}
      >
        {ChapterTitles.AVG}
      </PageHeading>
      <PageContent>
        <p>Hier ziet u een overzicht van uw ingediende AVG verzoeken.</p>
        <p>
          <Linkd external={true} href="https://www.amsterdam.nl/privacy/loket/">
            Loket persoonsgegevens gemeente Amsterdam
          </Linkd>
        </p>
      </PageContent>
      <SectionCollapsible
        id="SectionCollapsible-complaints"
        title="Lopende verzoeken"
        noItemsMessage="U heeft nog geen AVG verzoeken ingediend."
        startCollapsed={false}
        hasItems={!!avgVerzoekenLopend?.length}
        isLoading={isLoading(AVG)}
        className={styles.SectionCollapsibleFirst}
      >
        <Table items={avgVerzoekenLopend} displayProps={DISPLAY_PROPS_AVG} />
      </SectionCollapsible>
      <SectionCollapsible
        id="SectionCollapsible-complaints"
        title="Afgehandelde verzoeken"
        noItemsMessage="U hebt nog geen afgehandelde AVG verzoeken."
        startCollapsed={false}
        hasItems={!!avgVerzoekenAfgehandeld?.length}
        isLoading={isLoading(AVG)}
      >
        <Table
          items={avgVerzoekenAfgehandeld}
          displayProps={DISPLAY_PROPS_AVG}
        />
      </SectionCollapsible>
    </OverviewPage>
  );
};

export default AVG;
