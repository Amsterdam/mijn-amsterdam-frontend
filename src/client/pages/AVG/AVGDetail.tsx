import { generatePath } from 'react-router-dom';

import { AVGRequestFrontend } from '../../../server/services/avg/types';
import { AppRoutes } from '../../../universal/config/routes';
import { ThemaIcon } from '../../components';
import { Datalist } from '../../components/Datalist/Datalist';
import { ThemaTitles } from '../../config/thema';
import { useAVGDetailPage } from '../AVG/useAVGDetailPage.hook';
import ThemaDetailPagina from '../ThemaPagina/ThemaDetailPagina';

function getVerzoekRows(verzoek: AVGRequestFrontend) {
  return [
    { label: 'Nummer', content: verzoek.id },
    { label: 'Type verzoek', content: verzoek.type },
    { label: 'Onderwerp(en)', content: verzoek.themas },
    { label: 'Toelichting', content: verzoek.toelichting },
  ].filter((row) => row.content);
}

function AVGDetailContent({ verzoek }: { verzoek: AVGRequestFrontend }) {
  const rows = getVerzoekRows(verzoek);

  return <Datalist rows={rows} />;
}

function AVGDetail() {
  const { verzoek, isLoading, isError } = useAVGDetailPage();
  return (
    <ThemaDetailPagina<AVGRequestFrontend>
      title="AVG verzoek"
      zaak={verzoek}
      isError={isError}
      isLoading={isLoading}
      icon={<ThemaIcon />}
      pageContentTop={!!verzoek && <AVGDetailContent verzoek={verzoek} />}
      backLink={{
        to: generatePath(AppRoutes.AVG, {
          page: 1,
        }),
        title: ThemaTitles.AVG,
      }}
    />
  );
}

export default AVGDetail;
