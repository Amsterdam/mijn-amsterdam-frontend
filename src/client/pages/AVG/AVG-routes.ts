import { AVG } from './AVG';
import { AVGDetail } from './AVGDetail';
import { AVGList } from './AVGLijst';
import { FeatureToggle } from '../../../universal/config/feature-toggles';

export const AVGRoutes = [
  {
    route: '/avg/verzoek/:id',
    Component: AVGDetail,
    isActive: FeatureToggle.avgActive,
  },
  {
    route: '/avg/lijst/:kind/:page?',
    Component: AVGList,
    isActive: FeatureToggle.avgActive,
  },
  { route: '/avg', Component: AVG, isActive: FeatureToggle.avgActive },
];
