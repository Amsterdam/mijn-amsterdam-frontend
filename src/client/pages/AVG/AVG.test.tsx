import { render } from '@testing-library/react';
import { generatePath } from 'react-router';
import { MutableSnapshot } from 'recoil';

import { AppRoutes } from '../../../universal/config/routes';
import { appStateAtom } from '../../hooks/useAppState';
import MockApp from '../MockApp';
import { AVG } from './AVG';
import { testState } from './AVGDetail.test';
import { AVGResponse } from '../../../server/services/avg/types';
import { AppState } from '../../../universal/types';

function initializeState(testState: AppState) {
  return (snapshot: MutableSnapshot) => snapshot.set(appStateAtom, testState);
}

function setupTestComponent(testState: AppState) {
  const routeEntry = generatePath(AppRoutes.AVG);
  const routePath = AppRoutes.AVG;

  return function Component() {
    return (
      <MockApp
        routeEntry={routeEntry}
        routePath={routePath}
        component={AVG}
        initializeState={initializeState(testState)}
      />
    );
  };
}

describe('AVG thema pagina', () => {
  it('Matches the Full Page snapshot when there are requests', () => {
    const Component = setupTestComponent(testState);
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Matches the Full Page snapshot when there are no requests', () => {
    const themaContent: AVGResponse = {
      verzoeken: [],
      aantal: 0,
    };
    const Component = setupTestComponent({
      AVG: {
        status: 'OK',
        content: themaContent,
      },
    } as unknown as AppState);

    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Matches the Full Page snapshot when API gives no result', () => {
    const Component = setupTestComponent({
      AVG: {
        status: 'ERROR',
        content: null,
      },
    } as unknown as AppState);
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
