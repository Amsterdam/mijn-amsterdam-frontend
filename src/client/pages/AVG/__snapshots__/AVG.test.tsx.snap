// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AVG thema pagina > Matches the Full Page snapshot when API gives no result 1`] = `
<DocumentFragment>
  <main
    class="_Page_47d7c0 _OverviewPage_47d7c0 _AVG_7ac078"
  >
    <header
      class="_PageHeading_3b337d _hasIcon_3b337d _hasBackLink_3b337d"
    >
      <span
        aria-hidden="true"
        class="_Icon_3b337d"
      >
        <svg
          aria-label="AVG persoonsgegevens"
          class="_ChapterIcon_3ea2dd"
          fill="black"
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0 9C0 13.9706 4.02945 18 9 18V32H32V4H16.4845C14.8699 1.58801 12.1204 0 9 0C4.02945 0 0 4.02942 0 9ZM18 9C18 9.68738 17.9229 10.3568 17.777 11H26V13H17.0645C16.7026 13.7281 16.2451 14.4003 15.7083 15H26V17H15V15.7083C14.4003 16.2451 13.7281 16.7026 13 17.0645V28H28V8H17.9451C17.9814 8.32831 18 8.66199 18 9ZM15 19H26V21H15V19ZM26 23H15V25H26V23ZM9 3C7.34315 3 6 4.34315 6 6V8H5V14H13V8H12V6C12 4.34315 10.6569 3 9 3ZM9 5C8.44772 5 8 5.44772 8 6V8H10V6C10 5.44772 9.55228 5 9 5Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <div
        class="_HeadingInner_3b337d"
      >
        <a
          class="_Button_c3ab45 _Button__plain_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45 _BackLink_3b337d"
          href="/"
        >
          <svg
            aria-hidden="true"
            class="_Icon_c3ab45 _Icon__left_c3ab45"
            fill="#000"
            focusable="false"
            height="14px"
            viewBox="0 0 32 32"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-links-icoon
            </title>
            <polygon
              points="22.8571429 32 6.85714286 16 22.8571429 0 25.7568858 2.91002571 12.6771943 16 25.7568858 29.0899743"
            />
          </svg>
          Home
        </a>
        <h2
          class="amsterdam-heading amsterdam-heading--1"
        >
          AVG persoonsgegevens
        </h2>
      </div>
    </header>
    <div
      class="_PageContent_47d7c0"
    >
      <p>
        Hier ziet u een overzicht van uw ingediende AVG verzoeken.
      </p>
      <p>
        <a
          class="_Button_c3ab45 _Button__plain_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45"
          href="https://www.amsterdam.nl/privacy/loket/"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="_Icon_c3ab45 _Icon__left_c3ab45"
            fill="#000"
            focusable="false"
            height="14px"
            viewBox="0 0 32 32"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-rechts-icoon
            </title>
            <polygon
              points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
            />
          </svg>
          Loket persoonsgegevens gemeente Amsterdam
        </a>
      </p>
    </div>
    <section
      class="_SectionCollapsible_b93dc7 _SectionCollapsibleFirst_7ac078"
    >
      <h3
        class="amsterdam-heading amsterdam-heading--3 _Title_b93dc7 _TitleWithItems_b93dc7"
      >
        <button
          aria-expanded="true"
          class="_TitleToggle_b93dc7"
        >
          <svg
            aria-hidden="true"
            class="_CaretIcon_b93dc7"
            focusable="false"
            height="32px"
            viewBox="0 0 32 32"
            width="32px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-rechts-icoon
            </title>
            <polygon
              points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
            />
          </svg>
           Lopende verzoeken
        </button>
      </h3>
      <div
        aria-hidden="false"
        class="_Panel_b93dc7"
        style="height: 0px;"
      >
        <div
          class="_PanelInner_b93dc7"
        >
          <p
            class="_NoItemsMessage_b93dc7"
          >
            U heeft nog geen AVG verzoeken ingediend.
          </p>
        </div>
      </div>
    </section>
    <section
      class="_SectionCollapsible_b93dc7"
    >
      <h3
        class="amsterdam-heading amsterdam-heading--3 _Title_b93dc7 _TitleWithItems_b93dc7"
      >
        <button
          aria-expanded="true"
          class="_TitleToggle_b93dc7"
        >
          <svg
            aria-hidden="true"
            class="_CaretIcon_b93dc7"
            focusable="false"
            height="32px"
            viewBox="0 0 32 32"
            width="32px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-rechts-icoon
            </title>
            <polygon
              points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
            />
          </svg>
           Afgehandelde verzoeken
        </button>
      </h3>
      <div
        aria-hidden="false"
        class="_Panel_b93dc7"
        style="height: 0px;"
      >
        <div
          class="_PanelInner_b93dc7"
        >
          <p
            class="_NoItemsMessage_b93dc7"
          >
            U hebt nog geen afgehandelde AVG verzoeken.
          </p>
        </div>
      </div>
    </section>
  </main>
</DocumentFragment>
`;

exports[`AVG thema pagina > Matches the Full Page snapshot when there are no requests 1`] = `
<DocumentFragment>
  <main
    class="_Page_47d7c0 _OverviewPage_47d7c0 _AVG_7ac078"
  >
    <header
      class="_PageHeading_3b337d _hasIcon_3b337d _hasBackLink_3b337d"
    >
      <span
        aria-hidden="true"
        class="_Icon_3b337d"
      >
        <svg
          aria-label="AVG persoonsgegevens"
          class="_ChapterIcon_3ea2dd"
          fill="black"
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0 9C0 13.9706 4.02945 18 9 18V32H32V4H16.4845C14.8699 1.58801 12.1204 0 9 0C4.02945 0 0 4.02942 0 9ZM18 9C18 9.68738 17.9229 10.3568 17.777 11H26V13H17.0645C16.7026 13.7281 16.2451 14.4003 15.7083 15H26V17H15V15.7083C14.4003 16.2451 13.7281 16.7026 13 17.0645V28H28V8H17.9451C17.9814 8.32831 18 8.66199 18 9ZM15 19H26V21H15V19ZM26 23H15V25H26V23ZM9 3C7.34315 3 6 4.34315 6 6V8H5V14H13V8H12V6C12 4.34315 10.6569 3 9 3ZM9 5C8.44772 5 8 5.44772 8 6V8H10V6C10 5.44772 9.55228 5 9 5Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <div
        class="_HeadingInner_3b337d"
      >
        <a
          class="_Button_c3ab45 _Button__plain_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45 _BackLink_3b337d"
          href="/"
        >
          <svg
            aria-hidden="true"
            class="_Icon_c3ab45 _Icon__left_c3ab45"
            fill="#000"
            focusable="false"
            height="14px"
            viewBox="0 0 32 32"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-links-icoon
            </title>
            <polygon
              points="22.8571429 32 6.85714286 16 22.8571429 0 25.7568858 2.91002571 12.6771943 16 25.7568858 29.0899743"
            />
          </svg>
          Home
        </a>
        <h2
          class="amsterdam-heading amsterdam-heading--1"
        >
          AVG persoonsgegevens
        </h2>
      </div>
    </header>
    <div
      class="_PageContent_47d7c0"
    >
      <p>
        Hier ziet u een overzicht van uw ingediende AVG verzoeken.
      </p>
      <p>
        <a
          class="_Button_c3ab45 _Button__plain_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45"
          href="https://www.amsterdam.nl/privacy/loket/"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="_Icon_c3ab45 _Icon__left_c3ab45"
            fill="#000"
            focusable="false"
            height="14px"
            viewBox="0 0 32 32"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-rechts-icoon
            </title>
            <polygon
              points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
            />
          </svg>
          Loket persoonsgegevens gemeente Amsterdam
        </a>
      </p>
    </div>
    <section
      class="_SectionCollapsible_b93dc7 _SectionCollapsibleFirst_7ac078"
    >
      <h3
        class="amsterdam-heading amsterdam-heading--3 _Title_b93dc7 _TitleWithItems_b93dc7"
      >
        <button
          aria-expanded="true"
          class="_TitleToggle_b93dc7"
        >
          <svg
            aria-hidden="true"
            class="_CaretIcon_b93dc7"
            focusable="false"
            height="32px"
            viewBox="0 0 32 32"
            width="32px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-rechts-icoon
            </title>
            <polygon
              points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
            />
          </svg>
           Lopende verzoeken
        </button>
      </h3>
      <div
        aria-hidden="false"
        class="_Panel_b93dc7"
        style="height: 0px;"
      >
        <div
          class="_PanelInner_b93dc7"
        >
          <p
            class="_NoItemsMessage_b93dc7"
          >
            U heeft nog geen AVG verzoeken ingediend.
          </p>
        </div>
      </div>
    </section>
    <section
      class="_SectionCollapsible_b93dc7"
    >
      <h3
        class="amsterdam-heading amsterdam-heading--3 _Title_b93dc7 _TitleWithItems_b93dc7"
      >
        <button
          aria-expanded="true"
          class="_TitleToggle_b93dc7"
        >
          <svg
            aria-hidden="true"
            class="_CaretIcon_b93dc7"
            focusable="false"
            height="32px"
            viewBox="0 0 32 32"
            width="32px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-rechts-icoon
            </title>
            <polygon
              points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
            />
          </svg>
           Afgehandelde verzoeken
        </button>
      </h3>
      <div
        aria-hidden="false"
        class="_Panel_b93dc7"
        style="height: 0px;"
      >
        <div
          class="_PanelInner_b93dc7"
        >
          <p
            class="_NoItemsMessage_b93dc7"
          >
            U hebt nog geen afgehandelde AVG verzoeken.
          </p>
        </div>
      </div>
    </section>
  </main>
</DocumentFragment>
`;

exports[`AVG thema pagina > Matches the Full Page snapshot when there are requests 1`] = `
<DocumentFragment>
  <main
    class="_Page_47d7c0 _OverviewPage_47d7c0 _AVG_7ac078"
  >
    <header
      class="_PageHeading_3b337d _hasIcon_3b337d _hasBackLink_3b337d"
    >
      <span
        aria-hidden="true"
        class="_Icon_3b337d"
      >
        <svg
          aria-label="AVG persoonsgegevens"
          class="_ChapterIcon_3ea2dd"
          fill="black"
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0 9C0 13.9706 4.02945 18 9 18V32H32V4H16.4845C14.8699 1.58801 12.1204 0 9 0C4.02945 0 0 4.02942 0 9ZM18 9C18 9.68738 17.9229 10.3568 17.777 11H26V13H17.0645C16.7026 13.7281 16.2451 14.4003 15.7083 15H26V17H15V15.7083C14.4003 16.2451 13.7281 16.7026 13 17.0645V28H28V8H17.9451C17.9814 8.32831 18 8.66199 18 9ZM15 19H26V21H15V19ZM26 23H15V25H26V23ZM9 3C7.34315 3 6 4.34315 6 6V8H5V14H13V8H12V6C12 4.34315 10.6569 3 9 3ZM9 5C8.44772 5 8 5.44772 8 6V8H10V6C10 5.44772 9.55228 5 9 5Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <div
        class="_HeadingInner_3b337d"
      >
        <a
          class="_Button_c3ab45 _Button__plain_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45 _BackLink_3b337d"
          href="/"
        >
          <svg
            aria-hidden="true"
            class="_Icon_c3ab45 _Icon__left_c3ab45"
            fill="#000"
            focusable="false"
            height="14px"
            viewBox="0 0 32 32"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-links-icoon
            </title>
            <polygon
              points="22.8571429 32 6.85714286 16 22.8571429 0 25.7568858 2.91002571 12.6771943 16 25.7568858 29.0899743"
            />
          </svg>
          Home
        </a>
        <h2
          class="amsterdam-heading amsterdam-heading--1"
        >
          AVG persoonsgegevens
        </h2>
      </div>
    </header>
    <div
      class="_PageContent_47d7c0"
    >
      <p>
        Hier ziet u een overzicht van uw ingediende AVG verzoeken.
      </p>
      <p>
        <a
          class="_Button_c3ab45 _Button__plain_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45"
          href="https://www.amsterdam.nl/privacy/loket/"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="_Icon_c3ab45 _Icon__left_c3ab45"
            fill="#000"
            focusable="false"
            height="14px"
            viewBox="0 0 32 32"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-rechts-icoon
            </title>
            <polygon
              points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
            />
          </svg>
          Loket persoonsgegevens gemeente Amsterdam
        </a>
      </p>
    </div>
    <section
      class="_SectionCollapsible_b93dc7 _SectionCollapsibleFirst_7ac078"
    >
      <h3
        class="amsterdam-heading amsterdam-heading--3 _Title_b93dc7 _TitleWithItems_b93dc7"
      >
        <button
          aria-expanded="true"
          class="_TitleToggle_b93dc7"
        >
          <svg
            aria-hidden="true"
            class="_CaretIcon_b93dc7"
            focusable="false"
            height="32px"
            viewBox="0 0 32 32"
            width="32px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-rechts-icoon
            </title>
            <polygon
              points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
            />
          </svg>
           Lopende verzoeken
        </button>
      </h3>
      <div
        aria-hidden="false"
        class="_Panel_b93dc7"
        style="height: 0px;"
      >
        <div
          class="_PanelInner_b93dc7"
        >
          <table
            class="_Table_a66dcc"
          >
            <thead>
              <tr
                class="_TableRow_a66dcc"
              >
                <th
                  class="_DisplayProp_a66dcc"
                >
                  Nummer
                </th>
                <th
                  class="_DisplayProp_a66dcc"
                >
                  Ontvangen op
                </th>
                <th
                  class="_DisplayProp_a66dcc"
                >
                  Onderwerp
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="_TableRow_a66dcc"
              >
                <td
                  class="_DisplayProp_a66dcc"
                >
                  <span
                    class="_DisplayPropLabel_a66dcc"
                  >
                    Nummer:
                  </span>
                  <a
                    class="_Button_c3ab45 _Button__plain_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45"
                    href="/avg/verzoek/1"
                  >
                    <svg
                      aria-hidden="true"
                      class="_Icon_c3ab45 _Icon__left_c3ab45"
                      fill="#000"
                      focusable="false"
                      height="14px"
                      viewBox="0 0 32 32"
                      width="14px"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <title>
                        chevron-rechts-icoon
                      </title>
                      <polygon
                        points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
                      />
                    </svg>
                    1
                  </a>
                </td>
                <td
                  class="_DisplayProp_a66dcc"
                >
                  <span
                    class="_DisplayPropLabel_a66dcc"
                  >
                    Ontvangen op:
                  </span>
                  <span>
                    06 maart 2023
                  </span>
                </td>
                <td
                  class="_DisplayProp_a66dcc"
                >
                  <span
                    class="_DisplayPropLabel_a66dcc"
                  >
                    Onderwerp:
                  </span>
                  <span>
                    avg thema 3
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </section>
    <section
      class="_SectionCollapsible_b93dc7"
    >
      <h3
        class="amsterdam-heading amsterdam-heading--3 _Title_b93dc7 _TitleWithItems_b93dc7"
      >
        <button
          aria-expanded="true"
          class="_TitleToggle_b93dc7"
        >
          <svg
            aria-hidden="true"
            class="_CaretIcon_b93dc7"
            focusable="false"
            height="32px"
            viewBox="0 0 32 32"
            width="32px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-rechts-icoon
            </title>
            <polygon
              points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
            />
          </svg>
           Afgehandelde verzoeken
        </button>
      </h3>
      <div
        aria-hidden="false"
        class="_Panel_b93dc7"
        style="height: 0px;"
      >
        <div
          class="_PanelInner_b93dc7"
        >
          <table
            class="_Table_a66dcc"
          >
            <thead>
              <tr
                class="_TableRow_a66dcc"
              >
                <th
                  class="_DisplayProp_a66dcc"
                >
                  Nummer
                </th>
                <th
                  class="_DisplayProp_a66dcc"
                >
                  Ontvangen op
                </th>
                <th
                  class="_DisplayProp_a66dcc"
                >
                  Onderwerp
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="_TableRow_a66dcc"
              >
                <td
                  class="_DisplayProp_a66dcc"
                >
                  <span
                    class="_DisplayPropLabel_a66dcc"
                  >
                    Nummer:
                  </span>
                  <a
                    class="_Button_c3ab45 _Button__plain_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45"
                    href="/avg/verzoek/223"
                  >
                    <svg
                      aria-hidden="true"
                      class="_Icon_c3ab45 _Icon__left_c3ab45"
                      fill="#000"
                      focusable="false"
                      height="14px"
                      viewBox="0 0 32 32"
                      width="14px"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <title>
                        chevron-rechts-icoon
                      </title>
                      <polygon
                        points="9.75688579 32 6.85714286 29.0899743 19.9368344 16 6.85714286 2.91002571 9.75688579 0 25.7568858 16"
                      />
                    </svg>
                    223
                  </a>
                </td>
                <td
                  class="_DisplayProp_a66dcc"
                >
                  <span
                    class="_DisplayPropLabel_a66dcc"
                  >
                    Ontvangen op:
                  </span>
                  <span>
                    09 maart 2022
                  </span>
                </td>
                <td
                  class="_DisplayProp_a66dcc"
                >
                  <span
                    class="_DisplayPropLabel_a66dcc"
                  >
                    Onderwerp:
                  </span>
                  <span>
                    avg thema 4, avg thema 1
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </section>
  </main>
</DocumentFragment>
`;
